### ISSUE TITLE
color contrast accessibility issues

### ISSUE THREAD

**Issue by o-t-w** [ionic-team/ionic#17710]
I checked some of the default color with a color contrast tool https://webaim.org/resources/contrastchecker/

Some of them fail to reach an AA WCAG level of contrast. 

<img width="294" alt="screenshot 2019-03-07 at 13 40 15" src="https://user-images.githubusercontent.com/29547922/53960598-92b89b00-40de-11e9-9218-b060fd67a514.png">

**Comment by brandyscarney** [comment]
Thank you for the issue! 

While we understand your concern with the accessibility problems, it is not as simple as fixing the chip component as it is set up to match the spec and our default colors are used throughout multiple components.

 We provide this set of colors as a quick and easy way to style certain components, but we can't cover every use case for accessibility while matching the spec. How you use the colors and which colors those are is completely up to you. 

For example, our light color will **always** fail as text on a light background:

<img width="739" alt="screen shot 2019-03-07 at 1 31 45 pm" src="https://user-images.githubusercontent.com/6577830/53980901-7cf7a580-40df-11e9-8922-e477542f0263.png">

However, if you were to use it with a darker theme, it would pass fine:

<img width="746" alt="screen shot 2019-03-07 at 1 31 54 pm" src="https://user-images.githubusercontent.com/6577830/53980902-7cf7a580-40df-11e9-9fae-9efb2ba14e82.png">

I don't want you to think we don't care about accessibility because we do, but our goal is to provide a set of default colors and not control how or where you use them, while allowing them to be fully customized. 

Since chips are set up to follow the [Material Design guideline](https://material.io/go/design-chips), they have to have a background color that is equal to `0.08` opacity of the foreground color. Without being able to calculate the background color based on hue, it's hard for us to be able to change this value based on the lightness of the foreground. 

I updated our default colors with a few colors chosen by the [Material Design color tool](https://material.io/tools/color/) and you can see that the generated styles by some of these colors also fail for accessibility:

[Here it is in a Codepen](https://codepen.io/brandyscarney/pen/BbWbvW?editors=1100)

<img width="747" alt="screen shot 2019-03-07 at 1 41 09 pm" src="https://user-images.githubusercontent.com/6577830/53981020-c647f500-40df-11e9-883d-ad3d8c33f5f4.png">

<img width="770" alt="screen shot 2019-03-07 at 1 55 10 pm" src="https://user-images.githubusercontent.com/6577830/53981500-a95ff180-40e0-11e9-8697-0cd6d5f18f9d.png">

Meanwhile, I compared our tertiary color against the shade used in their [chip demo](https://material-components.github.io/material-components-web-catalog/#/component/chips) and they both pass AA but fail AAA.

Theirs:

<img width="746" alt="screen shot 2019-03-07 at 10 37 59 am" src="https://user-images.githubusercontent.com/6577830/53968612-68a6af00-40c5-11e9-8422-06fa1408ab21.png">

Ours:

<img width="739" alt="screen shot 2019-03-07 at 10 38 57 am" src="https://user-images.githubusercontent.com/6577830/53968649-70feea00-40c5-11e9-9b71-ed76416140da.png">

Notice if I update this same demo to use a different primary color taken from the Material Design colors, the result will fail the accessibility test:

<img width="1373" alt="screen shot 2019-03-07 at 1 57 10 pm" src="https://user-images.githubusercontent.com/6577830/53981681-107da600-40e1-11e9-8072-0902698738a5.png">

I hope you can understand that the colors should be used however you choose but we can't possibly cover every situation to assure accessibility. Please let me know if you have any questions or concerns, thank you!
